modules/algebrasAndModel/src/main/scala/utils/Types.scala

```diff
@@ -41,5 +44,37 @@ object Types {
   final case class Owner(owner: String)                             extends AnyVal
   final case class Repo(repo: String)                               extends AnyVal
   final case class Contributions(contributions: Int)                extends AnyVal
+  final case class Organization(organization: String)               extends AnyVal
+  object Organization {
+    implicit val organizationEncoder: Encoder[Organization] = deriveUnwrappedEncoder[Organization]
+    implicit val organizationDecoder: Decoder[Organization] = deriveUnwrappedDecoder[Organization]
+  }
+  final case class Repository(repository: String) extends AnyVal
+  object Repository {
+    implicit val repositoryDecoder: Decoder[Repository] = deriveUnwrappedDecoder[Repository]
+    implicit val repositoryEncoder: Encoder[Repository] = deriveUnwrappedEncoder[Repository]
+  }
+  final case class SubscribeAt(subscribeAt: String) extends AnyVal
   final case class Commit(sha: Sha, url: Url)
+  final case class GetSubscriptionData(
+      organization: Organization,
+      repository: Repository,
+      subscribeAt: SubscribeAt
+  )
+  final case class PostSubscriptionData(organization: Organization, repository: Repository)
+  object PostSubscriptionData {
+    implicit val postSubscriptionsDataDecoder: Decoder[PostSubscriptionData] =
+      deriveDecoder[PostSubscriptionData]
+    implicit val postSubscriptionsDataEncoder: Encoder[PostSubscriptionData] =
+      deriveEncoder[PostSubscriptionData]
+  }
+  final case class GetSubscriptions(subscriptions: List[GetSubscriptionData])
+  final case class PostSubscriptions(subscriptions: List[PostSubscriptionData])
+
+  object PostSubscriptions {
+    implicit val postSubscriptionsDecoder: Decoder[PostSubscriptions] =
+      deriveDecoder[PostSubscriptions]
+    implicit val postSubscriptionsEncoder: Encoder[PostSubscriptions] =
```

Comment:
You can now name those just as `decoder` and `encoder` since they don't interfere between them now
```suggestion
    implicit val encoder: Encoder[PostSubscriptions] =
```

Reply:
sure will do thanks